ramp_custom_lwfa/predictions.py

```diff
@@ -38,5 +38,30 @@ def combine(cls, predictions_list, index_list=None):
                 filtered_predictions = combined_predictions
 
             return cls(y_pred=np.array(filtered_predictions, dtype=object))
+        
+        @property
+        def valid_indexes(self):
+            """Return valid indexes, handling empty predictions."""
+            try:
+                if len(self.y_pred.shape) > 1:
+                    # If shape is (n,0), handle as empty predictions
+                    if self.y_pred.shape[1] == 0:
+                        return np.ones(self.y_pred.shape[0], dtype=bool)
+                    # If shape is (n,1) or any other wrong shape, raise error
+                    else:
+                        raise ValueError(
+                            f"Predictions have wrong shape {self.y_pred.shape}. "
+                            "Expected 1D array of lists, got 2D array. "
+                            "Each prediction should be a list of detections."
+                        )
```

Comment:
Why not simply raise the error (the empty prediction are returned in the except)